src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingTownHall.java

```diff
@@ -85,6 +93,86 @@ public BuildingEntry getBuildingRegistryEntry()
         return ModBuildings.townHall;
     }
 
+    @Override
+    public void onColonyTick(final IColony colony)
+    {
+        super.onColonyTick(colony);
+
+        List<IGlobalResearch> removes = new ArrayList<>();
+        for(IGlobalResearch research : MinecoloniesAPIProxy.getInstance().getGlobalResearchTree().getAutostartResearches())
```

Comment:
I believe Mekle001's more extreme use case was someone intending to thoroughly rework the trees, to such extent where the University itself was locked; having a way to start research outside of the University thus becomes necessary.  Outside of that, I think it's fairly plausible to see a more progression-oriented modpack use the research trees such that they'd have low or no research start costs, but intend to have a lot of low-tier research that's intended to be started as soon as it was unlocked.

You're probably right in the Town Hall not being the right place for it; there's a lot of early-game where you might not even have a constructed town hall, and maybe not even have the town hall block placed.  Maybe Colony itself?

Reply:
So, I see several places where onColonyTick is fired from: Citizens through CitizenManager, Visitors through visitorManager, buildings through BuidingManager, workorders through workManager, and colonyEvents through eventManager.

Finally, you could do it directly in worldTickSlow for the colony. I'm going to suggest that is where we should do this work, as it will handle the case where the townhall block is broken and not yet placed again.  The colony doesn't exist until it's created from the town hall, which is why I originally suggested there, but I had forgotten about the broken townhall case.

Reply:
Is it okay if I add an onColonyTick to ResearchManager and attach it to the worldTicksSlow in colony?  That seems like the logical place, and should reduce clutter in since ResearchManager is a lot less crowded than Colony right now.

Reply:
That seems like a fantastic solution.